docs/data/toolpad/concepts/data-providers.md

```diff
@@ -0,0 +1,111 @@
+# Data Providers
+
+<p class="description">Bring tabular data to the frontend with server-side pagination and filtering.</p>
+
+Toolpad functions are great to bring some backend state to the page, but they fall short when it comes to offering pagination and filtering capabilities from the server. Toolpad offers a special construct to enable this use case: Data providers. Data providers abstract server-side collections. They could be database tables, or REST APIs, or any data that represents a set of records that share a common interface. Data providers are defined as server-side objects and can be directly connected to a data grid to make it fully interactive.
+
+Follow these steps to create a new data provider
+
+1. Drag a data grid into the canvas
+
+2. Under its **Row Source** property, select the option **Data Provider**.
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/concepts/data-providers/rows-source.png", "alt": "Select data provider row source", "caption": "Select data provider row source", "zoom": false, "width": 297}}
+
+3. Click the data provider selector and choose **Create new data provider**.
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/concepts/data-providers/create-data-provider.png", "alt": "Create data provider", "caption": "Create data provider", "zoom": false, "width": 294}}
+
+4. Name the new data provider and click **Create**
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/concepts/data-providers/create-data-provider-dialog.png", "alt": "Create data provider dialog", "caption": "Create data provider dialog", "zoom": false, "width": 490}}
+
+5. Use the code button to open your code editor with the data provider backend.
+
+{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/concepts/data-providers/open-editor.png", "alt": "Open data provider editor", "caption": "Open data provider editor", "zoom": false, "width": 272}}
+
+6. A data provider that iterates over a static list could look as follows:
+
+   ```tsx
+   import { createDataProvider } from '@mui/toolpad-core/server';
+   import DATA from './movies.json';
+
+   export default createDataProvider({
+     async getRecords({ paginationModel: { start = 0, pageSize } }) {
+       const records = DATA.slice(start, start + pageSize);
+       return { records, totalCount: DATA.length };
+     },
+   });
+   ```
+
+## pagination
```

Comment:
```suggestion
## Pagination
```

In my opinion, is better to capitalize the features, as they reference to the functionality, not a specific prop/code equivalent.